### PR TITLE
Fix creating Credentials from API

### PR DESCRIPTION
**Pull request by agrare**

### CODE REVIEW

File: app/models/manageiq/providers/workflows/automation_manager/credential.rb [hunk 1]
```diff
@@ -62,8 +62,8 @@ class ManageIQ::Providers::Workflows::AutomationManager::Credential < ManageIQ::
   }.freeze
 
   def self.params_to_attributes(params)
-    allowed_params     = API_ATTRIBUTES.pluck(:id)
-    unpermitted_params = params.keys - allowed_params
+    allowed_params     = API_ATTRIBUTES.pluck(:id) + ["type"]
```
Comment (agrare) [inline app/models/manageiq/providers/workflows/automation_manager/credential.rb:65]: There is also a "type" holding the class name

Comment (agrare) [reply app/models/manageiq/providers/workflows/automation_manager/credential.rb:65]: No problem validating this, but is that something we do elsewhere?  I don't recall seeing that sort of check on other API payloads.  We wouldn't be in this class if the type wasn't already a subclass of this one

Comment (agrare) [reply app/models/manageiq/providers/workflows/automation_manager/credential.rb:65]: Actually no, there's no way we could get here if `"type"` wasn't already a valid subclass of this one https://github.com/ManageIQ/manageiq-api/blob/master/lib/services/api/authentication_service.rb#L4-L6

**Comment by kbrock** [comment]
fixed by https://github.com/ManageIQ/manageiq/pull/22570

**Pull request closed without merging**